apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/RestResourceBase.java

```diff
@@ -203,9 +267,22 @@ protected static byte[] decodeBase64(String base64encoded) {
 
   // // // Helper methods for JAX-RS response construction
 
-  protected static Response restServiceError(Response.Status httpStatus, String failMessage) {
-    return Response.status(httpStatus)
-        .entity(new ApiError(failMessage, httpStatus.getStatusCode()))
-        .build();
+  protected static Uni<RestResponse<Object>> apiErrorResponseUni(
+      Response.Status httpStatus, String failMessage) {
+    return Uni.createFrom().item(apiErrorResponse(httpStatus, failMessage));
+  }
+
+  protected static RestResponse<Object> apiErrorResponse(
+      Response.Status httpStatus, String failMessage) {
+    return RestResponse.status(httpStatus, new ApiError(failMessage, httpStatus.getStatusCode()));
+  }
```

Comment:
Are those needed? Usually an exception mapper would do the transformation?

Reply:
In most cases, yes, but I think there was one or two cases where there is more local information available to give better error message.

Reply:
That's why we have our own exception, where we can push custom messages and then have a way to catch that exception and transform to the `ApiError`.. Just saying this is an option as well, ass you can always push a failure down on the reactive sequence..

Reply:
Did end up removing and using `WebApplicationException`